controllers/etcdcluster_controller.go

```diff
@@ -162,6 +163,24 @@ func (r *EtcdClusterReconciler) addNewMember(ctx context.Context, cluster *etcdv
 	return &reconcilerevent.MemberAddedEvent{Object: cluster, Member: member, Name: peerName}, nil
 }
 
+type MembersByName []etcdclient.Member
+
+func (a MembersByName) Len() int           { return len(a) }
+func (a MembersByName) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a MembersByName) Less(i, j int) bool { return a[i].Name < a[j].Name }
+
+func (r *EtcdClusterReconciler) removeMember(ctx context.Context, cluster *etcdv1alpha1.EtcdCluster, members []etcdclient.Member) (*reconcilerevent.MemberRemovedEvent, error) {
```

Comment:
Document this function and consider also looking at the leadership status of the member that we choose to remove....and at least logging when we are about to remove a leader.